### PR TITLE
Number Convertions Binary to Octal in Java#865

### LINKED ISSUE #865: Number Convertions in different languages
Can i work on it ??

### PR DESCRIPTION
**Pull request by mruna7**
Fixes #865 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added a file binary to octal.java
- 

#### Languages Used:
- Java
- 

#### Files Added:
- Binary_to_octal.java
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by mruna7** [comment]
@asha15  mam I request you  please review this.

**Comment by mruna7** [comment]
> @asha15 mam I request you please review this.

@Vaibhavraj10  @asha15 @jainaman224  please review this request

**Comment by asha15** [comment]
please complete the PR template

**Comment by mruna7** [comment]
> please complete the PR template

yes !!

File: Number_Conversion/Java/Binary_To_Octal.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	       decimal += temp * Math.pow(2, n);  
+	       num = num / 10;  
+	       n++;  
+	  }
+	    
+	  int octal[] = new int[20];
+	  int i = 0;
+	  while(decimal > 0)
+	  {
+              int t = decimal % 8;
```
Comment (singh-shreya6) [inline Number_Conversion/Java/Binary_To_Octal.java:21]: Please correct indentation here.

File: Number_Conversion/Java/Binary_To_Octal.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	      octal[i++] = t;
+       	      decimal = decimal / 8;
```
Comment (singh-shreya6) [inline Number_Conversion/Java/Binary_To_Octal.java:23]: Please correct indentation here.

File: Number_Conversion/Java/Binary_To_Octal.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import java.io.*; 
+import java.util.Scanner;
+public class Binary_To_Octal
```
Comment (singh-shreya6) [inline Number_Conversion/Java/Binary_To_Octal.java:3]: Please add one blank line here.

File: Number_Conversion/Java/Decimal_to_Binary [hunk 4]
```diff
@@ -0,0 +1,42 @@
+/**
+ * Decimal To Binary Conversion
+ */
+package algo_ds;
+
+import java.util.Scanner;
+
+/**
```
Comment (singh-shreya6) [inline Number_Conversion/Java/Decimal_to_Binary:8]: Please remove author tag

File: Number_Conversion/Java/Decimal_to_Binary [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ */
+public class DecimalToBinary
+{
+            public static void Decimal_to_binary(int decimal)
+	{   
+	        int i = 0;
+		int binary[] = new int[100];
+		while(decimal > 0)
+		{
+			binary[i] = decimal % 2;
```
Comment (singh-shreya6) [inline Number_Conversion/Java/Decimal_to_Binary:20]: Please correct indentation here.

File: Number_Conversion/Java/Decimal_to_Binary [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+import java.util.Scanner;
+
+/**
+ * @author mrunal
+ *
+ */
+public class DecimalToBinary
+{
+            public static void Decimal_to_binary(int decimal)
```
Comment (singh-shreya6) [inline Number_Conversion/Java/Decimal_to_Binary:14]: Please correct indentation here.

File: Number_Conversion/Java/Decimal_to_Binary [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+			decimal = decimal / 2;
+			i++;
+		}
+		System.out.println("binary representation");
+		for(int j = i; j <= i; j--)
+		    System.out.print(binary[j]);
+	}
+	public static void main(String[] args)
+	{
+		// TODO Auto-generated method stub
```
Comment (singh-shreya6) [inline Number_Conversion/Java/Decimal_to_Binary:30]: Please correct indentation here.

File: Number_Conversion/Java/Decimal_to_Binary [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+		Scanner sc = new Scanner(System.in);
+		System.out.println("Enter a decimal number");
+		int decimal = sc.nextInt();
+		Decimal_to_binary(decimal);
+	}
+}
+/*
+Input:
+4
```
Comment (singh-shreya6) [inline Number_Conversion/Java/Decimal_to_Binary:39]: Input should have the statement Enter a decimal number

File: Number_Conversion/Java/Binary_To_Octal.java [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+          Scanner sc = new Scanner(System.in);
+          System.out.println("enter Binary number");
+          num = sc.nextLong();
+          binToOctal(num);  
+          sc.close();
+      }
+}
+/*
+Input:
+1011001010
```
Comment (singh-shreya6) [inline Number_Conversion/Java/Binary_To_Octal.java:41]: Here it should have Enter a Binary Number

File: Number_Conversion/Java/Binary_To_Octal.java [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+	  }
+	  System.out.print("Octal number:");
+	  for(int j = i-1 ; j >= 0 ; j--)
+	      System.out.print(octal[j]); 
+      } 
+      public static void main(String[] args)
+      {
+          long num;
+          Scanner sc = new Scanner(System.in);
+          System.out.println("enter Binary number");
```
Comment (singh-shreya6) [inline Number_Conversion/Java/Binary_To_Octal.java:33]: Change it to Enter a Binary Number

**Comment by mruna7** [comment]
> Can you please split the different languages code as part of different PR. You will be scored accordingly.

mam I have written the code in java only.

**Comment by singh-shreya6** [comment]
@mruna7 I mean to say that split the two codes into two different PRs. You will get more score that way. One PR, one code.

**Comment by singh-shreya6** [comment]
@mruna7 I just checked the repository, decimal to binary code already exists in Java. Please remove that. Just keep Binary to Octal in this PR.

**Comment by mruna7** [comment]
> @mruna7 I just checked the repository, decimal to binary code already exists in Java. Please remove that. Just keep Binary to Octal in this PR.

yes @singh-shreya6 mam ,done!!

**Comment by mruna7** [comment]
> Looks good @mruna7. Please squash your commits now so that it can be merged.

mam .I am not able to squash the commits .Can you please help me out?

**Comment by mruna7** [comment]
> > Looks good @mruna7. Please squash your commits now so that it can be merged.

@shreyakapoor08  mam .Done!!

**Pull request merged**